src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEMKernel.hpp

```diff
@@ -321,6 +321,355 @@ struct DampingMatrixKernel
+        }
+      } );
+
+
+      //Step 2: Initial iteration of (M^{-1}K)p
+      pAuxView.zero();
+      forAll< EXEC_POLICY >( sizeElem, [=] GEOS_HOST_DEVICE ( localIndex const k )
+      {
+
+        real64 xLocal[numNodesPerElem][3];
+        for( localIndex a=0; a< numNodesPerElem; ++a )
+        {
+          for( localIndex i=0; i<3; ++i )
+          {
+            xLocal[a][i] = X( elemsToNodes( k, a ), i );
+          }
+        }
+
+        real32 auxx[numNodesPerElem]  = {0.0};
+        real32 auyy[numNodesPerElem]  = {0.0};
+        real32 auzz[numNodesPerElem]  = {0.0};
+        real32 uelemx[numNodesPerElem] = {0.0};
+
+
+        // Volume integration
+        for( localIndex q=0; q<numQuadraturePointsPerElem; ++q )
+        {
+
+          m_finiteElement.computeFirstOrderStiffnessTermX( q, xLocal, [&] ( int i, int j, real32 dfx1, real32 dfx2, real32 dfx3 )
+          {
+            auxx[i] -= dfx1*uxView[k][j];
+            auyy[i] -= dfx2*uyView[k][j];
+            auzz[i] -= dfx3*uzView[k][j];
+          } );
+
+          m_finiteElement.computeFirstOrderStiffnessTermY( q, xLocal, [&] ( int i, int j, real32 dfy1, real32 dfy2, real32 dfy3 )
+          {
+            auxx[i] -= dfy1*uxView[k][j];
+            auyy[i] -= dfy2*uyView[k][j];
+            auzz[i] -= dfy3*uzView[k][j];
+          } );
+
+          m_finiteElement.computeFirstOrderStiffnessTermZ( q, xLocal, [&] ( int i, int j, real32 dfz1, real32 dfz2, real32 dfz3 )
+          {
+            auxx[i] -= dfz1*uxView[k][j];
+            auyy[i] -= dfz2*uyView[k][j];
+            auzz[i] -= dfz3*uzView[k][j];
+          } );
+
+        }
+
+        //Time update + multiplication by inverse of the mass matrix
+        for( localIndex i = 0; i < numNodesPerElem; ++i )
+        {
+          real32 diag=(auxx[i]+auyy[i]+auzz[i]);
+          uelemx[i]+=diag;
+
+          real32 const localIncrement = uelemx[i]/mass[elemsToNodes[k][i]];
+          RAJA::atomicAdd< ATOMIC_POLICY >( &pAuxView[elemsToNodes[k][i]], localIncrement );
+        }
+
+      } );
+
+
+
+      lambdaOld = lambdaNew;
+
+      dotProductPPaux = 0.0;
+      normP=0.0;
+      WaveSolverUtils::dotProduct( sizeNode, pView, pAuxView, dotProductPPaux );
+      WaveSolverUtils::dotProduct( sizeNode, pView, pView, normP );
+
+      lambdaNew = dotProductPPaux/normP;
+
+      normPaux = 0.0;
+      WaveSolverUtils::dotProduct( sizeNode, pAuxView, pAuxView, normPaux );
+
+      forAll< EXEC_POLICY >( sizeNode, [=] GEOS_HOST_DEVICE ( localIndex const a )
+      {
+        pView[a] = pAuxView[a]/( normPaux );
+      } );
+
+      if( abs( lambdaNew-lambdaOld )/abs( lambdaNew )<= epsilon )
+      {
+        counter++;
+      }
+      else
+      {
+        counter=0;
+      }
+
+      numberIter++;
+
+
+    }
+    while (counter < 10 && numberIter < nIterMax);
+
+    GEOS_THROW_IF( numberIter> nIterMax, "Power Iteration algorithm does not converge", std::runtime_error );
+
+    real64 dt = 1.99/sqrt( abs( lambdaNew ));
```

Comment:
1.99 because you are using a 0.5% margin with respect to the CFL condition? Please add a small comment to explain